### ISSUE TITLE
Future Perfect and Imp Weird Error

### ISSUE THREAD

**Issue by finnbar** [mtgred/netrunner#1844]
Okay, so I ran R&D with an Imp, and encountered a Future Perfect, which I Imped. On Imping it, the psigame happened, which was fine. I lost this psigame, and then another psigame triggered? And then I won that psigame, so Jinteki Personal Evolution fired. And then it fired again. And it kept firing. I have no idea how this happened, but it's slightly crazy.

Also, looking online, it seems that in a real game I'd be able to Imp it after failing the psigame rather than before. Is there any way to do this on Jinteki.net? Would it be possible to add a trash step for on-access Psi games? This sounds like a nightmare to implement though.

Stack Trace:
```clojure
#error {
 :cause nil
 :via
 [{:type java.lang.NullPointerException
   :message nil
   :at [clojure.lang.Numbers ops "Numbers.java" 1013]}]
 :trace
 [[clojure.lang.Numbers ops "Numbers.java" 1013]
  [clojure.lang.Numbers gt "Numbers.java" 229]
  [clojure.lang.Numbers gt "Numbers.java" 3864]
  [game.core$steal$fn__5034 invoke "core-runs.clj" 50]
  [game.core$effect_completed invokeStatic "core-events.clj" 224]
  [game.core$effect_completed invoke "core-events.clj" 220]
  [game.core$trigger_event_simult$fn__3902$fn__3903$fn__3904 invoke "core-events.clj" 177]
  [game.core$effect_completed invokeStatic "core-events.clj" 224]
  [game.core$effect_completed invoke "core-events.clj" 220]
  [game.core$trigger_event_simult_player$choose_handler__3837$fn__3846$fn__3847 invoke "core-events.clj" 96]
  [game.core$effect_completed invokeStatic "core-events.clj" 224]
  [game.core$effect_completed invoke "core-events.clj" 220]
  [game.core$resolve_damage$fn__4126 invoke "core-rules.clj" 174]
  [game.core$effect_completed invokeStatic "core-events.clj" 224]
  [game.core$effect_completed invoke "core-events.clj" 220]
  [game.core$trigger_event_sync$fn__3828 invoke "core-events.clj" 64]
  [game.core$effect_completed invokeStatic "core-events.clj" 224]
  [game.core$effect_completed invoke "core-events.clj" 220]
  [game.core$trigger_event_sync_next invokeStatic "core-events.clj" 52]
  [game.core$trigger_event_sync_next doInvoke "core-events.clj" 39]
  [clojure.lang.RestFn applyTo "RestFn.java" 157]
  [clojure.core$apply invokeStatic "core.clj" 654]
  [clojure.core$apply doInvoke "core.clj" 641]
  [clojure.lang.RestFn invoke "RestFn.java" 587]
  [game.core$trigger_event_sync invokeStatic "core-events.clj" 63]
  [game.core$trigger_event_sync doInvoke "core-events.clj" 54]
  [clojure.lang.RestFn invoke "RestFn.java" 580]
  [game.core$resolve_damage invokeStatic "core-rules.clj" 159]
  [game.core$resolve_damage invoke "core-rules.clj" 153]
  [game.core$damage$fn__4137 invoke "core-rules.clj" 204]
  [game.core$resolve_prompt invokeStatic "core-actions.clj" 145]
  [game.core$resolve_prompt invoke "core-actions.clj" 115]
  [game.main$handle_do invokeStatic "main.clj" 75]
  [game.main$handle_do invoke "main.clj" 72]
  [game.main$handle_command invokeStatic "main.clj" 147]
  [game.main$handle_command invoke "main.clj" 137]
  [game.main$run$fn__23464 invoke "main.clj" 173]
  [game.main$run invokeStatic "main.clj" 170]
  [game.main$run invoke "main.clj" 163]
  [game.main$_main$fn__23480 invoke "main.clj" 226]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.lang.Thread run "Thread.java" 745]]}
```

**Comment by Saintis** [comment]
Interesting. So am I understanding that the PE trigger repeated itself, dealing multiple sets of 1 net damage? Or did the psi-game repeatedly fire (after the first two mentioned)?

Also, at what point did the error appear?

The stack trace suggests that a non-existing agenda was "stolen" (or at least something with `nil` points).

Will do some investigating.

**Comment by JoelCFC25** [comment]
This one is very old--originally in at #1137.

**Comment by finnbar** [comment]
The error appeared on succeeding the psi-game - I was then asked to 
"prevent 1 net damage?" (I had Deus X installed) multiple times, but did 
/close-prompt to avoid dying.
It was definitely more times than the psi-game was played, though, so 
it's not some weird thing where by succeeding one psi-game I ended up 
stealing four Future Perfects or whatever. :p

The error appeared on the psi-game success, I believe.

Thanks! :)

**Comment by NoahTheDuke** [comment]
Closed via #3271.
